tools/clang/test/HLSLFileCheck/passes/transforms/reassociate/reassociation-flag.hlsl

```diff
@@ -0,0 +1,31 @@
+// RUN: %dxc -T cs_6_3 -E cs_main %s -opt-enable enable-aggressive-reassociation  | FileCheck %s -check-prefixes=CHECK,COMMON_FACTOR
```

Comment:
This is good for validating that the flag gets passed through from the driver, but I think we should also add a way to control it for unit tests. 

To add it for unit testing we need to modify the hctdb.py file. This will hook it up for using in dxcopt. You can see an example of how it is done for the instcombine [NoSink ](https://github.com/microsoft/DirectXShaderCompiler/blob/d9caef5289dc7fee304817963ea55d229a1eae4d/utils/hct/hctdb.py#L6500)flag

```
        add_pass(
            "instcombine",
            "InstructionCombiningPass",
            "Combine redundant instructions",
            [
                {"n": "NoSink", "t": "bool", "c": 1},
            ],
        )
```

And then how it is used for testing [tools/clang/test/HLSLFileCheck/passes/llvm/instcombine/nosink.ll](https://github.com/microsoft/DirectXShaderCompiler/blob/d9caef5289dc7fee304817963ea55d229a1eae4d/tools/clang/test/HLSLFileCheck/passes/llvm/instcombine/nosink.ll#L1)

Then we can take the unit test from the previous PR and run it with dxopt passing the flag on and off to show the difference. You will have to put the test in a new file under HLSLFileCheck since that is how we run a dxopt vs. opt test.

Reply:
Done.